pennylane_lightning/src/simulator/Measures.hpp

```diff
@@ -157,6 +159,37 @@ class Measures {
         return std::real(expected_value);
     };
 
+    /**
+     * @brief Expected value of a Sparse Hamiltonian.
+     *
+     * @param row_map_ptr   row_map array pointer.
+     *                      The j element encodes the number of non-zeros above
+     * row j.
+     * @param row_map_size  row_map array size.
+     * @param entries_ptr   pointer to an array with column indices of the
+     * non-zero elements.
+     * @param values_ptr    pointer to an array with the non-zero elements.
+     * @param numNNZ        number of non-zero elements.
+     * @return fp_t
+     */
+    template <class index_type>
+    fp_t expval(const index_type *row_map_ptr, const index_type row_map_size,
+                const index_type *entries_ptr, const CFP_t *values_ptr,
+                const index_type numNNZ) {
```

Comment:
Not sure if this is a good idea as I don't know how much you need to use such long lists of inputs. But, do you think that encapsulating these inputs (e.g., in a struct) would make this function signature cleaner? It not, forget my comment :slightly_smiling_face: